tests/php/ORM/DataObjectDuplicationTest.php

```diff
@@ -2,29 +2,27 @@
 
 namespace SilverStripe\ORM\Tests;
 
-use SilverStripe\ORM\DataObject;
 use SilverStripe\Dev\SapphireTest;
-use SilverStripe\ORM\FieldType\DBDatetime;
 
 class DataObjectDuplicationTest extends SapphireTest
 {
+    protected static $fixture_file = 'DataObjectDuplicationtest.yml';
```

Comment:
I find this test a lot harder to read with 37 lines of YAML fixtures for one big unit test. Mainly because the classes are just named sequentially (class1 to class7), rather than having a name that would imply their relation to each other. But that's just a continuation of the existing test, so I can't fault you for that :) Overall, think just creating the various associations in PHP based on YAML fixtures would improve readability quite a bit.

Reply:
Ok now they are all animals.

Ask again and I'm making them chemical molecules. :)